### PR TITLE
Deprecate "namespace" in telemetry calls

### PR DESCRIPTION
**Pull request by rli**
"Namespace" as a concept doesn't mesh well with our team's new naming strategy for telemetry events. This change marks all references to "namespace" as deprecated and allows building telemetry events that are not tied to a top-level "namespace".

In the future, we should:
* have all of our metric names in a schematized format that can be shared across our products.
* remove the `ActionWrappers` and `LoggingDialogWrapper` system
* migrate the existing telemetry names into the new format

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: core/src/software/aws/toolkits/core/telemetry/MetricEvent.kt [hunk 1]
```diff
@@ -10,13 +10,15 @@ import java.time.Instant
 import software.amazon.awssdk.services.toolkittelemetry.model.Unit as MetricUnit
 
 interface MetricEvent {
-    val namespace: String
+    @Deprecated("namespace should not be used")
```
Comment (rli) [inline core/src/software/aws/toolkits/core/telemetry/MetricEvent.kt]: Alternatively, should we deprecate this whole file and have the TelemetryService record individual `Datum` that are not encapsulated by a `MetricEvent`?

File: build.gradle [hunk 2]
```diff
@@ -136,9 +136,6 @@ subprojects {
         tasks.withType(JavaCompile).all {
             options.compilerArgs << "-Werror"
         }
-        tasks.withType(KotlinCompile).all {
```
Comment (hunterwerlla) [inline build.gradle]: Was this accidentally removed?

**Comment by bryceitoc9** [comment]
I'm assuming this won't affect any existing metric names? If so, would we just not be providing namespaces to new metrics?

File: build.gradle (hunk 2 above)
Comment (rli) [reply build.gradle]: No, since deprecation result in warnings we can no longer fail the build

**Comment by rli** [comment]
> I'm assuming this won't affect any existing metric names? If so, would we just not be providing namespaces to new metrics?

That's correct

File: core/src/software/aws/toolkits/core/telemetry/MetricEvent.kt (hunk 1 above)
Comment (awschristou) [reply core/src/software/aws/toolkits/core/telemetry/MetricEvent.kt]: What does it mean to have more than one datum with the same createTime?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/DefaultTelemetryPublisher.kt [hunk 3]
```diff
@@ -58,9 +58,12 @@ class DefaultTelemetryPublisher(
     private fun Collection<MetricEvent>.toMetricData(): Collection<MetricDatum> = this
         .flatMap { metricEvent ->
             metricEvent.data.map { datum ->
+                val metricName = metricEvent.namespace?.let {
+                    "$it.${datum.name}"
```
Comment (awschristou) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/DefaultTelemetryPublisher.kt]: `it.` or `it_` ?

File: core/src/software/aws/toolkits/core/telemetry/MetricEvent.kt (hunk 1 above)
Comment (hunterwerlla) [reply core/src/software/aws/toolkits/core/telemetry/MetricEvent.kt]: Is there a drawback of recording individual Datum?

Comment (rli) [reply core/src/software/aws/toolkits/core/telemetry/MetricEvent.kt]: Data with the same createTime results in each datum with the same timestamp when recorded in the back-end. I see the whole class as a convenience, so there is no downside in recording datum individually.

Comment (awschristou) [reply core/src/software/aws/toolkits/core/telemetry/MetricEvent.kt]: I'm not seeing the value in sending Datum with a matching timestamp. I'm neutral on the initial question.

Comment (rli) [reply core/src/software/aws/toolkits/core/telemetry/MetricEvent.kt]: If we're planning on eventually codegen-ing our metric classes, we're going to throw this whole file away anyways, so I'm inclined to merge as-is.

**Pull request merged**